### DISCUSSION TITLE
Removing things that developer mode downloaded/installed

### DISCUSSION THREAD

**Discussion by injust** [Homebrew/discussions#5504] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1c98a33a376e5b35ebc801d0f7d6ba7b9eb118ac
Last commit: 11 days ago
Core tap JSON: 18 Jul 05:43 UTC
Core cask tap JSON: 18 Jul 05:43 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: micro
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.45.2 => /usr/local/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Putting non-prefixed findutils in your path can cause python builds to fail.

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  fish_add_path /usr/local/sbin
```

### Description of issue

Previously, I wanted to send a PR for a Homebrew formula/cask, so I followed the instructions for contributions, which enabled developer mode. I've already untapped everything it tapped, and disabled developer mode.

But if I recall correctly, it also automatically downloaded/installed some other things, which I probably no longer need. How can I figure out what was downloaded/installed and clean it all up?

**Comment by SMillerDev** [comment]
> How can I figure out what was downloaded/installed and clean it all up?

There isn't really a defined list of what developer commands will install. It heavily depends on what's on your machine, what command you ran, when you ran it, what the thing was that you ran the command on and possibly some more factors.
Essentially the only way is a complete reinstall of brew or removing and restoring files until only the ones that are absolutely necessary remain... neither of which are likely worth the effort since those downloads don't actually do anything without developer mode.

**Reply by injust** [reply]
> Essentially the only way is a complete reinstall of brew or removing and restoring files until only the ones that are absolutely necessary remain

Incidentally, nuking `/usr/local/Homebrew/` and reinstalling was surprisingly painless.

**Comment by carlocab** [comment, marked as answer]
You can try
```
brew install-bundler-gems --groups=
```
to uninstall the development gems. I don't think there's anything else you need to get rid of.

**Reply by injust** [reply]
Yup I think this was it. Thank you!
